github/VisualStudio
# Signal completion when opening repository using Get to Code

When the `Get to Code` flow calls `AcquireCodeContainerAsync`, it uses `downloadProgress` completion as a signal to close the dialog. This PR ensures that this signal is sent when opening a repository as well as cloning a repository.

### What this PR does

- Adds tests for the `GitHub.StartPage` project / `GitHubContainerProvider` class
- Signal `downloadProgress` as complete before `AcquireCodeContainerAsync` returns a `CodeContainer`

It uses the following call to signal completion (1 of 1 steps complete with no progress message):
```cs
downloadProgress.Report(new ServiceProgressData(string.Empty, string.Empty, 1, 1));
```

### How to test

1. Install a version of Visual Studio 2019 that includes `Get to Code` + `GitHub` integration
2. Open Visual Studio 2019
3. Install Dogfood VSIX + restart VS
4. Click `Tools > Dogfood` and select .vsix artifact (or URL) for this PR + restart VS
5. Click `Clone or checkout code`
6. Click on `GitHub` button under `Browse a repository`
7. Select a repository + local path that already exists
8. Click on `Open` button
9. Expect `Clone or checkout code` window to close
10. Expect your repository to be visible on `Team Explorer` tool window

Fixes #2211
